Jenkinsfile

```diff
@@ -33,30 +33,34 @@ slack = new Slack()
 //
 // Smoke tests are CI-friendly test configuration.  Currently-supported Python version + modern C*/DSE instances.
 // We also avoid cython since it's tested as part of the nightlies.
+DEFAULT_CASSANDRA = ['2.1', '2.2', '3.0', '3.11', '4.0']
+DEFAULT_DSE = ['dse-5.0.15', 'dse-5.1.35', 'dse-6.0.18', 'dse-6.7.17', 'dse-6.8.30']
+DEFAULT_RUNTIME = ['2.7.18', '3.5.9', '3.6.10', '3.7.7', '3.8.3']
```

Comment:
It's a little strange to keep 2.7.x + 3.5.x + 3.6.x around here.  My rationale is that this PR is really about removing the special handling of "develop" branches + some normalization for how versions are specified.  Actual changes in versions we test against will be done in a PR explicitly focused on that role (and should be accompanied by corresponding changes to the runners).

Reply:
This is a great way to keep these versions in a single place, TIL!